.github/workflows/bundle-tracking.yml

```diff
@@ -0,0 +1,59 @@
+name: Bundle Size Tracking
+
+on:
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
+
+jobs:
+  track-bundle-size:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@08eba0b27e820071cde6df949e0beb9ba4906955 # ratchet:actions/checkout@v4
```

Comment:
Seems like some of this boilerplate setup could be extracted to a template and used in multiple workflows.